helm s3 ingress templating throwing `wrong type for value; expected string; got map[string]interface {}`

### Describe the bug
https://github.com/seaweedfs/seaweedfs/pull/5463 has introduced a bug for the S3 ingress annotation. After the latest version of the helm chart (`3.65.0`) I get this in the logs: 

Failure Log example:

> Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = Manifest generation error (cached): `helm template . --name-template nextcloud-seaweedfs --namespace nextcloud --kube-version 1.29 --values /tmp/14146e97-0d68-491a-bdbf-864746d34c6a <api versions removed> --include-crds` failed exit status 1: coalesce.go:289: warning: destination for seaweedfs.s3.ingress.annotations is a table. Ignoring non-table value () Error: template: seaweedfs/templates/s3-ingress.yaml:15:19: executing "seaweedfs/templates/s3-ingress.yaml" at <.Values.s3.ingress.annotations>: wrong type for value; expected string; got map[string]interface {} Use --debug flag to render out invalid YAML

#### To Reproduce

Clone the current repo on the master branch and run:
```
# change to the correct directory
cd k8s/charts/seaweedfs
```
Then, try changing the values.yaml to have a standard annotation for `s3.ingress.annotations` (don't forget to set `s3.ingress.enabled` to `true`):
```yaml
s3:
  ingress:
    enabled: true
    annotations:
      nginx.ingress.kubernetes.io/proxy-body-size: 1G
```
Then run `helm template .`:
```console
helm template .
Error: template: seaweedfs/templates/s3-ingress.yaml:15:19: executing "seaweedfs/templates/s3-ingress.yaml" at <.Values.s3.ingress.annotations>: wrong type for value; expected string; got map[string]interface {}

Use --debug flag to render out invalid YAML
```

### System Setup
This is for the helm chart specifically and to do with ingress. I'm using the latest verison released here:
https://github.com/seaweedfs/seaweedfs/releases/tag/3.65

## Expected behavior
The issue is that the correct type is not list `[]` or string `""`.
The correct type for annotations is a map `{}`.

You can verify this by creating a new helm chart with:
```bash
helm create test-chart
```

Then inspect the default `annotations` in values.yaml and then the deployment template:
```console
$ grep annotations test/values.yaml
  annotations: {}
  annotations: {}
  
$ grep annotations -A2 -B1 test-chart/templates/deployment.yaml
      {{- with .Values.podAnnotations }}
      annotations:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

You can see another example at the nextcloud helm chart [here](https://github.com/nextcloud/helm/blob/30c69c16298c189dd78c930660a17a2b5b774773/charts/nextcloud/values.yaml#L28)

This is the standard so that users can then provide a map in the values.yaml like so:

```yaml
annotations:
  cert-manager.io/cluster-issuer: 'letsencrypt-staging'
```

I'll submit a PR to fix the helm chart.

--- a/k8s/charts/seaweedfs/Chart.yaml
+++ b/k8s/charts/seaweedfs/Chart.yaml
@@ -2,4 +2,4 @@ apiVersion: v1
 description: SeaweedFS
 name: seaweedfs
 appVersion: "3.65"
-version: 3.65.0
+version: 3.65.1

--- a/k8s/charts/seaweedfs/templates/filer-ingress.yaml
+++ b/k8s/charts/seaweedfs/templates/filer-ingress.yaml
@@ -11,9 +11,9 @@ kind: Ingress
 metadata:
   name: ingress-{{ template "seaweedfs.name" . }}-filer
   namespace: {{ .Release.Namespace }}
-  {{- if .Values.filer.ingress.annotations }}
+  {{- with .Values.filer.ingress.annotations }}
   annotations:
-    {{ tpl .Values.filer.ingress.annotations . | nindent 4 | trim }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
   labels:
     app.kubernetes.io/name: {{ template "seaweedfs.name" . }}

--- a/k8s/charts/seaweedfs/templates/master-ingress.yaml
+++ b/k8s/charts/seaweedfs/templates/master-ingress.yaml
@@ -11,9 +11,9 @@ kind: Ingress
 metadata:
   name: ingress-{{ template "seaweedfs.name" . }}-master
   namespace: {{ .Release.Namespace }}
-  {{- if .Values.master.ingress.annotations }}
+  {{- with .Values.master.ingress.annotations }}
   annotations:
-    {{ tpl .Values.master.ingress.annotations . | nindent 4 | trim }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
   labels:
     app.kubernetes.io/name: {{ template "seaweedfs.name" . }}

--- a/k8s/charts/seaweedfs/templates/s3-ingress.yaml
+++ b/k8s/charts/seaweedfs/templates/s3-ingress.yaml
@@ -10,9 +10,9 @@ kind: Ingress
 metadata:
   name: ingress-{{ template "seaweedfs.name" . }}-s3
   namespace: {{ .Release.Namespace }}
-  {{- if .Values.s3.ingress.annotations }}
+  {{- with .Values.s3.ingress.annotations }}
   annotations:
-    {{- tpl .Values.s3.ingress.annotations . | nindent 4 }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
   labels:
     app.kubernetes.io/name: {{ template "seaweedfs.name" . }}

--- a/k8s/charts/seaweedfs/values.yaml
+++ b/k8s/charts/seaweedfs/values.yaml
@@ -171,7 +171,7 @@ master:
     className: "nginx"
     # host: false for "*" hostname
     host: "master.seaweedfs.local"
-    annotations: |
+    annotations:
       nginx.ingress.kubernetes.io/auth-type: "basic"
       nginx.ingress.kubernetes.io/auth-secret: "default/ingress-basic-auth-secret"
       nginx.ingress.kubernetes.io/auth-realm: 'Authentication Required - SW-Master'
@@ -540,7 +540,7 @@ filer:
     className: "nginx"
     # host: false for "*" hostname
     host: "seaweedfs.cluster.local"
-    annotations: |
+    annotations:
       nginx.ingress.kubernetes.io/backend-protocol: GRPC
       nginx.ingress.kubernetes.io/auth-type: "basic"
       nginx.ingress.kubernetes.io/auth-secret: "default/ingress-basic-auth-secret"
@@ -746,7 +746,7 @@ s3:
     # host: false for "*" hostname
     host: "seaweedfs.cluster.local"
     # additional ingress annotations for the s3 endpoint
-    annotations: ""
+    annotations: {}
     tls: []
 
 certificates:

